test/metabase/api/collection_test.clj

```diff
@@ -1155,6 +1155,150 @@
                      :authority_level nil
                      :entity_id       (:entity_id collection)
                      :can_write       true}])
+                 (for [item (:data (mt/user-http-request :crowberto :get 200 "collection/root/items"))]
+                   (dissoc item :id))))))
+
+      (testing "using a snippet without parameters is true"
+        (mt/with-temp* [NativeQuerySnippet [snippet {:content    "table"
+                                                     :creator_id (mt/user->id :crowberto)
+                                                     :name       "snippet"}]
+                        Card [card {:name          "Business Card"
+                                    :dataset_query {:native {:template-tags {:param0  {:required false
+                                                                                       :default  0}
+                                                                             :snippet {:name         "snippet"
+                                                                                       :type         :snippet
+                                                                                       :snippet-name "snippet"
+                                                                                       :snippet-id   (:id snippet)}}
+                                                             :query "select {{param0}} from {{snippet}}"}}}]]
+          (is (= (let [collection (collection/user->personal-collection (mt/user->id :crowberto))]
```

Comment:
Probably better to use `partial=` here. Otherwise if we add a new column to Collection and it comes back from `GET /api/collection/root/items` then we'd have to update this test. That sort of thing adds up -- before we had tooling like that in place we could have to update a hundred tests every time we added a new column.

We have `mt/object-defaults` as well which accomplishes a similar purpose if `partial=` won't work for one reason or another

Reply:
You also wouldn't have to loop over everything and `(dissoc item :id)` on every row if you used `parital=` -- you could just leave that out of the _expected_ form